components/new-contribution-flow/StepDetails.js

```diff
@@ -53,7 +53,7 @@ const StepDetails = ({ onChange, data, collective, tier, showFeesOnTop, isTaxDed
                 type="number"
                 min={1}
                 step={1}
-                max={tier.maxQuantity}
+                max={tier?.maxQuantity}
```

Comment:
Forgot this one, same thing we won't enter the `tier?.type` block if tier doesn't exist

```suggestion
                max={tier.maxQuantity}
```